wolfcrypt/src/asn.c

```diff
@@ -11598,14 +11634,24 @@ int PemToDer(const unsigned char* buff, long longSz, int type,
 }
 
 int wc_PemToDer(const unsigned char* buff, long longSz, int type,
-              DerBuffer** pDer, void* heap, EncryptedInfo* info, int* eccKey)
+              DerBuffer** pDer, void* heap, EncryptedInfo* info, int* keyFormat)
 {
-    return PemToDer(buff, longSz, type, pDer, heap, info, eccKey);
+    int ret = PemToDer(buff, longSz, type, pDer, heap, info, keyFormat);
+    if (ret == 0 && type == PRIVATEKEY_TYPE) {
+        DerBuffer* der = *pDer;
+        /* if a PKCS8 key header exists remove it */
+        ret = ToTraditional(der->buffer, der->length);
```

Comment:
Why are PKCS8 headers automatically removed here now?

Reply:
To maintain backwards compatibility on this public API. The `PemToDer` function does not strip PKCS8 headers now.